doc/self-hosting.md

```diff
@@ -123,7 +123,7 @@ Redis installation.
 Server secrets
 --------------
 
-You can add your own server secrets in `private/secret.json`.
+You can add your own server secrets in environment variables or `config/local.yml`.
```

Comment:
One comment I'd have here is that if you're a self-hosting user and you've got all your secrets set up in `private/secret.json`, upgrading to this revision or later is going to break your install :wrench:

I don't really have a clear idea how self-hosting users are managing their installs, but is there a way we can communicate this change well to a self-hosting user? As you've already flagged, one tool available is documentation, although maintaining a changelog when we don't really do releases/tags is a challenge. Maybe we should be doing releases too :thinking: . Perhaps we could also do something like in the bootstrap process could we check for the existence of `private/secret.json` and if it exists throw an exception with a message saying "you need to move your config to env vars/yaml"?

Also, if everyone has to re-write their config file anyway, maybe we should just drop this https://github.com/badges/shields/blob/5233a25ddcee9cc14eb8b67329aa019af613325e/services/jira/jira-base.js#L14-L25 in this PR as there isn't really any backwards-compatibility to preserve at this point.

Reply:
> I don't really have a clear idea how self-hosting users are managing their installs, but is there a way we can communicate this change well to a self-hosting user? As you've already flagged, one tool available is documentation, although maintaining a changelog when we don't really do releases/tags is a challenge. Maybe we should be doing releases too 🤔 . Perhaps we could also do something like in the bootstrap process could we check for the existence of `private/secret.json` and if it exists throw an exception with a message saying "you need to move your config to env vars/yaml"?

Yea, I'm a big fan of failing loudly. That's what's happening here: https://github.com/badges/shields/pull/2626/files#diff-e2e5bd472244fbfb4a40d38300d42534R6

This PR should make it possible for PaaS users to use the shields repo as-is, without forking or checking in any files. If we can solve that for Docker users, even better. Maybe we can publish an official docker image. Regardless, tagging releases probably would be a good idea.

Besides failing loudly and linking to a changelog, I'm not sure we have any other way to communicate this. The changelog will provide this answer going forward, though: it gives a single place to look when you're updating your self-hosted install.

I'd like to provide a second path for self-hosting someday, which is for advanced users who want to do something custom. We could ship an npm package with the server logic. Then the user can create their own project and use `shields-server` as a dependency. They can add their own services, and maybe even pick and choose which shields services should be enabled. This server package would be an ordinary library, and could follow semver. (Obviously this is not going to be tackled here.)

Reply:
> Yea, I'm a big fan of failing loudly. That's what's happening here: https://github.com/badges/shields/pull/2626/files#diff-e2e5bd472244fbfb4a40d38300d42534R6

oh right yes sorry - you've already done that exact thing and I missed it :man_facepalming: